### PR TITLE
Unit tests for condition steps in installer

### PR DESCRIPTION
**Pull request by nilsanderselde**
Increases `install` package test coverage from 20.1% to 25.6% of statements.

### CODE REVIEW

File: pkg/install/condition_test.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+			name:               "Status not complete",
+			configMapName:      "bootstrap",
+			configMapNamespace: "kube-system",
+		},
+		{
+			name:               "Bootstrap config map is ready",
+			configMapName:      "bootstrap",
+			configMapNamespace: "kube-system",
+			configMapStatus:    "complete",
+			want:               true,
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/install/condition_test.go]: `want` is `false` wherever it isn't explicitly set to `true`, so it is `false` in 3/4 of these cases. The lack of required values in the first 3 cases causes the errors you're talking about. I could make it more obvious by explicitly setting the values to something wrong.

File: pkg/install/condition_test.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+					Namespace: tt.configMapNamespace,
+				},
+				Data: map[string]string{
+					"status": tt.configMapStatus,
+				},
+			}),
+		}
+		ready, err := i.bootstrapConfigMapReady()
+		if err != nil {
+			t.Error(errMustBeNilMsg)
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/install/condition_test.go]: These condition functions are all called from within `wait.PollImmediateUntil` at https://github.com/Azure/ARO-RP/blob/8a51a88b572d1c9d1dd12f01d5d519b34bfedf51/pkg/install/install.go#L196

`PollImmediateUntil` https://github.com/kubernetes/apimachinery/blob/49ae5f6b1c038a611e9f98322f81f39125999bcf/pkg/util/wait/wait.go#L486 takes a `ConditionFunc` as an argument https://github.com/kubernetes/apimachinery/blob/49ae5f6b1c038a611e9f98322f81f39125999bcf/pkg/util/wait/wait.go#L206 which returns `bool, error`.

One question that could be asked is, "Are there any specific errors which should cause the installer to close the stop channel before the timeout?" Right now, these condition functions are all retried until the timeout, regardless of what happens. Maybe that is the right solution.

I put this specific check in this test (`err` must be `nil`) due to the current implementation. I was thinking specifically about regression testing, and somebody having the same question that you and I did: "Why is `err` always `nil`?"

File: pkg/install/condition_test.go [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+func TestAPIServersReady(t *testing.T) {
+	for _, tt := range []struct {
+		name                 string
+		serverName           string
+		availableCondition   operatorv1.ConditionStatus
+		progressingCondition operatorv1.ConditionStatus
+		want                 bool
+	}{
+		{
+			name: "Can't get Kubernetes API server",
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/install/condition_test.go]: Sure, that makes sense.

File: pkg/install/condition_test.go [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+func TestOperatorConsoleReady(t *testing.T) {
+	for _, tt := range []struct {
+		name               string
+		consoleName        string
+		availableCondition operatorv1.ConditionStatus
+		want               bool
+	}{
+		{
+			name: "Can't get Kubernetes API server",
```
[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/install/condition_test.go]: Copy/paste sloppiness. I'll fix it.

File: pkg/install/condition_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (m1kola) [reply pkg/install/condition_test.go]: Got it, thanks for the explanation.

File: pkg/install/condition_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (m1kola) [reply pkg/install/condition_test.go]: @jdhuntington sorry, my comment wasn't clear. I'm not talking about explicit or implicit `false` value in `want`.

We need an extra use case for each test to cover the situation when `Get` returns an error. For example, here:

https://github.com/Azure/ARO-RP/blob/6e694797696883e0ea0105150247ed009dfb9614/pkg/install/condition.go#L13-L16

We do cover line 15, but do not cover all the cases on line 15.

[the remark this answers is not in the document]
Comment (m1kola) [reply pkg/install/condition_test.go]: > The lack of required values in the first 3 cases causes the errors you're talking about.

Ah I think I now get what you mean. So when there is no configmap, `Get` will return an error. Is this what you mean?

[the remark this answers is not in the document]
Comment (nilsanderselde) [reply pkg/install/condition_test.go]: Yep. If I don't give it "bootstrap", it can't get it.

**Pull request merged**